.github/workflows/merge-check.yml

```diff
@@ -0,0 +1,32 @@
+name: Merge checks
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - '*'
+
+jobs:
+  gradle:
```

Comment:
Maybe call this `Build` instead? Or you might need the key here to be lowercase `build` and then set a name like `name: Build project`

https://docs.github.com/en/actions/quickstart